README.md

```diff
@@ -1,634 +1,1855 @@
+keep his business running! There is far less risk of an economic recession with this scenario. And
+it is all thanks to being able to compose DeFi protocols.
+
+DApp composability isn't just about risk-management, it is also one of the killer features of eUTxO
+blockchains. You can execute 15 different smart contracts in a single Cardano transaction; this
+probably equates to 5-7 different DApps being executed in this single transaction. This
+composability enables things in DeFi that people in TradFi can only dream about. Consider the
+example of converting DJED to ADA to buy an options contract on the secondary market only to
+immediately execute it; this all takes place in a single transaction.
+
+How would you do the above composition in TradFi? Just the first step of converting DJED to ADA
+would likely require a few days for the trade to settle. In the meantime, you are exposed to the
+volatility of ADA. Then, if your broker doesn't support the secondary market, you need to transfer
+your ADA to another broker which would take another few days to settle. Buying the options contract
+may take ANOTHER few days to settle. Do you see where this is going? By the time all of the settling
+has finished, the opportunity may have long passed.
+
+While the above example doesn't include Cardano-Loans, the point is that composition allows for
+creating truly complex economic transactions from very simple building blocks. Just like with
+Wolfram's cellular automaton, it is very hard to predict what the ultimate economy will look like.
+Because of this, **all core DeFi DApps should priortize composability**. People may come up with
+useful compositions that the DApp developers never considered. The more compositions that are
+possible, the larger, and more complex the DeFi economy can become. And as the risk-management
+example shows, composability can make the complex DeFi economy more resilient.
+
+### No Trustless Credit History
+
+You simply cannot have a healthy credit-debt market without credit history. However, if a market
+does have credit history but that history is not trustless, it just creates another attack vector
+governments and hackers can use to target the growing DeFi economy. 
+
+Some DApps try to rely on off-chain credit history databases, but this is not a good approach.
+*Reading* credit histories must be just as trustless as *writing* credit histories. Even if the
+blockchain records the credit history trustlessly, if lenders can't query the credit history, it is
+effectively the same as there being no credit history. A centralized database, or a database that is
+hard to read, is easy to commandeer/censor. 
+
+In practical terms, if only beefy computers are able to read the credit histories from the
+blockchain, those with these computers are in a priviledged position between borrowers and lenders.
+They can deny sharing the credit histories for borrowers they don't like. This would effectively
```

Comment:
They might even do the same for borrowers they DO like in order to stifle competition from other lenders.